src/preact/component/component.type.js

```diff
@@ -53,3 +53,32 @@ var RendererFunctionResponseType;
  * @typedef {function(!JsonObject):(?RendererFunctionResponseType|!Promise<?RendererFunctionResponseType>)}
  */
 var RendererFunctionType;
+
+/** @const */
+var IframeEmbedDef = {};
+
+/**
+ * @typedef {{
+ *   allow: (string|undefined),
+ *   allowFullScreen: (boolean|undefined),
+ *   allowTransparency: (boolean|undefined),
+ *   loading: (Loading),
```

Comment:
Unfortunately, this type cannot be transferred over to extern. I've been using `string` and a bit surprised it doesn't complain for you here.